test/node-api/test_should_return_to_caller/test_should_return_to_caller.c

```diff
@@ -0,0 +1,26 @@
+#define NAPI_EXPERIMENTAL
+#include <node_api.h>
+#include "../../js-native-api/common.h"
+
+static napi_value test(napi_env env, napi_callback_info info) {
```

Comment:
this loop doesn't terminate when using a watchdog and i'm not sure why

Reply:
I think this might have to do with N-API’s implicit `TryCatch`s around everything…?

Reply:
`HasTerminated` on the nearest TryCatch is false... is this a bug of some sort @nodejs/v8?